CRM/Contribute/Form/ContributionPage/Amount.php

```diff
@@ -337,6 +335,7 @@ public static function formRule($fields, $files, $self) {
     if (isset($fields['is_pay_later'])) {
       if (empty($fields['pay_later_text'])) {
         $errors['pay_later_text'] = ts('Please enter the text for the \'pay later\' checkbox displayed on the contribution form.');
+        $this->addElement('checkbox', 'is_monetary', ts('Execute real-time monetary transactions'));
```

Comment:
I don't think we should be creating contributions if the amount is $0 - but I suspect that could be open to debate. We definitely SHOULD be creating memberships though.

Whenever we fix something on the front end  contribution page we should add a test in this class api_v3_ContributionPageTest or for backoffice this class CRM_Contribute_Form_ContributionTest or the membership class.

Otherwise we can't expect this flakey code to stay fixed.